wagtail_extensions/blocks.py

```diff
@@ -93,10 +98,76 @@ def clean(self, value):
         return super().clean(value)
 
 
+class OpeningTimeBlock(blocks.StructBlock):
+    """
+    A semi-structured opening times block.
+
+    Period is left as a free text input to allow for
+    """
+    PUBLIC = 7
+    PUBLIC_LABEL = 'Public holiday'
+    WEEKDAYS = tuple(enumerate(calendar.day_name)) + ((PUBLIC, PUBLIC_LABEL),)
+
+    weekday = blocks.ChoiceBlock(choices=WEEKDAYS, required=False)
+    label = blocks.CharBlock(help_text='Optionally override default weekday label', required=False)
+    date = blocks.DateBlock(help_text='Optionally specify a day these times are for', required=False)
+    start = blocks.TimeBlock(required=False)
+    end = blocks.TimeBlock(required=False)
+    closed = blocks.BooleanBlock(default=False, required=False)
+
+    class Meta:
+        template = 'wagtail_extensions/blocks/openingtime.html'
+
+    def clean(self, value):
+        cleaned = super().clean(value)
+        start, end, weekday, date = map(cleaned.get, ['start', 'end', 'weekday', 'date'])
+        errors = defaultdict(ErrorList)
+
+        if date and date < datetime.date.today():
+            err = ValidationError('Dates need to be in the future')
```

Comment:
This will cause errors when saving an old block with old dates - not sure if that is necessarily a problem though. Alternative would be to silently discard old dates when saving?

Reply:
I think I would rather keep the explicit behaviour, rather than things just disappearing... also it would require more changes to the parent list block to be able to discard items.